multitenancy-spring-boot-autoconfigure/src/main/java/org/axonframework/extensions/multitenancy/autoconfig/MultiTenantEventScheduler.java

```diff
@@ -0,0 +1,268 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.axonframework.extensions.multitenancy.autoconfig;
+
+import org.axonframework.common.BuilderUtils;
+import org.axonframework.common.Registration;
+import org.axonframework.eventhandling.EventMessage;
+import org.axonframework.eventhandling.scheduling.EventScheduler;
+import org.axonframework.eventhandling.scheduling.ScheduleToken;
+import org.axonframework.extensions.multitenancy.components.MultiTenantAwareComponent;
+import org.axonframework.extensions.multitenancy.components.NoSuchTenantException;
+import org.axonframework.extensions.multitenancy.components.TargetTenantResolver;
+import org.axonframework.extensions.multitenancy.components.TenantDescriptor;
+import org.axonframework.extensions.multitenancy.components.eventstore.TenantEventSchedulerSegmentFactory;
+import org.axonframework.messaging.Message;
+
+import java.time.Duration;
+import java.time.Instant;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+import static org.axonframework.common.BuilderUtils.assertNonNull;
+
+/**
+ * Multi-tenant implementation of {@link EventScheduler} that delegates to a tenant-specific {@link EventScheduler}
+ * based on the {@link TenantDescriptor} resolved by the {@link TargetTenantResolver}.
+ *
+ * @author Stefan Dragisic
+ * @since 4.9.0
+ */
+public class MultiTenantEventScheduler implements EventScheduler, MultiTenantAwareComponent {
+
+    private final TenantEventSchedulerSegmentFactory tenantSegmentFactory;
+    private final TargetTenantResolver<EventMessage<?>> targetTenantResolver;
+
+    private final Map<TenantDescriptor, EventScheduler> tenantSegments = new ConcurrentHashMap<>();
+
+    /**
+     * Instantiate a {@link MultiTenantEventScheduler} based on the fields contained in the {@link Builder}.
+     * @param builder the {@link Builder} used to instantiate a {@link MultiTenantEventScheduler} instance
+     */
+    public MultiTenantEventScheduler(MultiTenantEventScheduler.Builder builder) {
+        builder.validate();
+        this.tenantSegmentFactory = builder.tenantSegmentFactory;
+        this.targetTenantResolver = builder.targetTenantResolver;
+    }
+
+    /**
+     * Instantiate a Builder to be able to create a {@link MultiTenantEventScheduler}.
+     * @return a Builder to be able to create a {@link MultiTenantEventScheduler}
+     */
+    public static MultiTenantEventScheduler.Builder builder() {
+        return new MultiTenantEventScheduler.Builder();
+    }
+
+    /**
+     * Schedule the given {@code event} for publication at the given {@code triggerDateTime}.
+     * Message must contain MetaData with {@link TenantDescriptor} to resolve the tenant.
+     * Therefor only EventMessage type is supported.
```

Comment:
This is quite a (understandable) deviation from the singular `EventSchedulers`. 
So, perhaps I would strengthen the sentence a bit further like so:
```suggestion
     * </p>
     * It is <em>required</em> that the given {@code event} is of type {@link EventMessage}, containing a resolvable {@link TenantDescriptor} from the {@link Message#getMetaData meta data}.
     * Without a {@code TenantDescriptor}, the `MultiTenantEventScheduler` is incapable of resolving the tenant-specific {@link EventScheduler}.
     * Therefor, the provided {@code event} should be of type {@code EventMessage} <em>with</em> a {@code TenantDescriptor} in it's {@link MetaData meta data}.
```

Reply:
I also think it's good if we specify this pointer in the class-level JavaDoc of this `EventScheduler` implementation.